tests/t0123_gateway_json_cbor_test.go

```diff
@@ -1,13 +1,858 @@
+						Header("Content-Type").
+							Not().Contains("application/%s", row.Format),
+					).Body(
+					formatedDir,
+				),
+			},
+			/**
+			test_expect_success "GET UnixFS as $name with 'Accept: application/vnd.ipld.dag-$format' converts to the expected Content-Type" '
+			curl -sD - -H "Accept: application/vnd.ipld.dag-$format" "http://127.0.0.1:$GWAY_PORT/ipfs/$FILE_CID" > curl_output 2>&1 &&
+			test_should_contain "Content-Disposition: ${disposition}\; filename=\"${FILE_CID}.${format}\"" curl_output &&
+			test_should_contain "Content-Type: application/vnd.ipld.dag-$format" curl_output &&
+			test_should_not_contain "Content-Type: application/$format" curl_output
+			'
+			*/
+			{
+				Name: fmt.Sprintf("GET UnixFS as %s with 'Accept: application/vnd.ipld.dag-%s' converts to the expected Content-Type", row.Name, row.Format),
+				Request: Request().
+					Path("ipfs/%s", fileCID).
+					Headers(
+						Header("Accept", "application/vnd.ipld.dag-%s", row.Format),
+					),
+				Response: Expect().
+					Status(200).
+					Headers(
+						Header("Content-Disposition").
+							Contains("%s; filename=\"%s.%s\"", row.Disposition, fileCID, row.Format),
+						Header("Content-Type").
+							Equals("application/vnd.ipld.dag-%s", row.Format),
+						Header("Content-Type").
+							Not().Contains("application/%s", row.Format),
+					),
+			},
+			/**
+			test_expect_success "GET UnixFS as $name with 'Accept: foo, application/vnd.ipld.dag-$format,bar' converts to the expected Content-Type" '
+			curl -sD - -H "Accept: foo, application/vnd.ipld.dag-$format,text/plain" "http://127.0.0.1:$GWAY_PORT/ipfs/$FILE_CID" > curl_output 2>&1 &&
+			test_should_contain "Content-Type: application/vnd.ipld.dag-$format" curl_output
+			'
+			*/
+			{
+				Name: fmt.Sprintf("GET UnixFS as %s with 'Accept: foo, application/vnd.ipld.dag-%s,bar' converts to the expected Content-Type", row.Name, row.Format),
+				Request: Request().
+					Path("ipfs/%s", fileCID).
+					Headers(
+						Header("Accept", "foo, application/vnd.ipld.dag-%s,bar", row.Format),
+					),
+				Response: Expect().
+					Status(200).
+					Headers(
+						Header("Content-Type").
+							Equals("application/vnd.ipld.dag-%s", row.Format),
+					),
+			},
+			/**
+			test_expect_success "GET UnixFS with format=$format (not dag-$format) is no-op (no conversion)" '
+			curl -sD headers "http://127.0.0.1:$GWAY_PORT/ipfs/$FILE_CID?format=$format" > curl_output 2>&1 &&
+			ipfs cat $FILE_CID > cat_output &&
+			test_cmp cat_output curl_output &&
+			test_should_contain "Content-Type: text/plain" headers &&
+			test_should_not_contain "Content-Type: application/$format" headers &&
+			test_should_not_contain "Content-Type: application/vnd.ipld.dag-$format" headers
+			'
+			*/
+			{
+				Name: fmt.Sprintf("GET UnixFS with format=%s (not dag-%s) is no-op (no conversion)", row.Format, row.Format),
+				Request: Request().
+					Path("ipfs/%s?format=%s", fileCID, row.Format),
```

Comment:
Is there are reason to use one version over another or are they interchangeable?
```suggestion
					Path("ipfs/%s", fileCID).
					Query("format", row.Format),
```